Source/WebCore/style/Styleable.cpp

```diff
@@ -893,7 +893,7 @@ void Styleable::updateCSSScrollTimelines(const RenderStyle* currentStyle, const
 
         for (auto& previousTimelineName : currentStyle->scrollTimelineNames()) {
             if (!currentTimelineNames.contains(previousTimelineName))
-                timelinesController->unregisterNamedScrollTimeline(previousTimelineName);
+                timelinesController->unregisterNamedTimeline(previousTimelineName, element);
```

Comment:
I wonder whether this is correct. Do we need to keep a separate method for `ScrollTimeline` and `ViewTimeline` so that we look up the timeline to remove by type? Is there any provision made in the spec for `scroll-timeline-name` and `view-timeline-name` to have exclusive names?

Reply:
From the spec:
> In case of a name conflict on the same element, names declared later in the naming property ([scroll-timeline-name](https://drafts.csswg.org/scroll-animations-1/#propdef-scroll-timeline-name), [view-timeline-name](https://drafts.csswg.org/scroll-animations-1/#propdef-view-timeline-name)) take precedence, and [scroll progress timelines](https://drafts.csswg.org/scroll-animations-1/#scroll-progress-timelines) take precedence over [view progress timelines](https://drafts.csswg.org/scroll-animations-1/#view-progress-timelines).

I think based on this we should never have an element with more than one timeline associated with the same name (although we will have to implement this logic in a later patch).